### PR TITLE
Only warn when specified server IP addresses don't match intf

### PR DESCRIPTION
**Pull request by MartinBasti**
In containers local addresses differ from public addresses and we need
a way to provide only public address to installers.

https://pagure.io/freeipa/issue/2715
https://pagure.io/freeipa/issue/4317

### CODE REVIEW

File: ipaserver/install/installutils.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def get_server_ip_address(host_name, unattended, setup_dns, ip_addresses):
     if len(hostaddr):
         for ha in hostaddr:
             try:
-                ips.append(ipautil.CheckedIPAddress(ha, match_local=True))
+                ips.append(ipautil.CheckedIPAddress(ha, match_local=False))
```
Comment (tomaskrizek) [inline ipaserver/install/installutils.py]: Shouldn't this behaviour also change on line 279?

**Comment by tomaskrizek** [comment]
Do we want to handle `ipa-client-install --ip-address` in the same way? I think the behaviour should be consistent.

File: ipaserver/install/installutils.py (hunk 1 above)
Comment (MartinBasti) [inline ipaserver/install/installutils.py]: Indeed it should

**Comment by MartinBasti** [comment]
I don't see reason why not, PR updated.

**Comment by tomaskrizek** [comment]
Thanks for the updates, looks good now.

**Comment by MartinBasti** [comment]
master:

* 6637980af6069623d944d9d592cbadba20b610a2 Only warn when specified server IP addresses don't match intf


ipa-4-5:

* 6206ac8bd23250bda0f8eb628f422671b9b99ad1 Only warn when specified server IP addresses don't match intf

**Pull request closed without merging**